lib/OrdinaryDiffEqBDF/src/algorithms.jl

```diff
@@ -1,11 +1,61 @@
-"""
-E. Alberdi Celayaa, J. J. Anza Aguirrezabalab, P. Chatzipantelidisc. Implementation of
-an Adaptive BDF2 Formula and Comparison with The MATLAB Ode15s. Procedia Computer Science,
-29, pp 1014-1026, 2014. doi: https://doi.org/10.1016/j.procs.2014.05.091
+function BDF_docstring(description::String,
+    name::String;
+    references::String = "",
+    extra_keyword_description::String = "",
+    extra_keyword_default::String = "")
+    keyword_default = """
+        chunk_size = Val{0}(),
+        autodiff = true,
+        standardtag = Val{true}(),
+        concrete_jac = nothing,
+        diff_type = Val{:forward},
+        linsolve = nothing,
+        precs = DEFAULT_PRECS,
+        """ * extra_keyword_default
+
+    keyword_default_description = """
+        - `chunk_size`: TBD
+        - `autodiff`: TBD
+        - `standardtag`: TBD
+        - `concrete_jac`: TBD
+        - `diff_type`: TBD
+        - `linsolve`: TBD
+        - `precs`: TBD
+        - `precs`: TBD
+        """ * extra_keyword_description
+
+    generic_solver_docstring(
+        description, name, "Multistep Method.", references,
+        keyword_default_description, keyword_default
+    )
+end
 
-ABDF2: Multistep Method
-An adaptive order 2 L-stable fixed leading coefficient multistep BDF method.
-"""
+
+@doc BDF_docstring("An adaptive order 2 L-stable fixed leading coefficient multistep BDF method.",
+    "ABDF2",
+    references = """
+    E. Alberdi Celayaa, J. J. Anza Aguirrezabalab, P. Chatzipantelidisc. Implementation of
+    an Adaptive BDF2 Formula and Comparison with The MATLAB Ode15s. Procedia Computer Science,
+    29, pp 1014-1026, 2014. doi: https://doi.org/10.1016/j.procs.2014.05.091
+    """,
+    extra_keyword_description = """
+    - `κ`: TBD
+    - `tol`: TBD
+    - `nlsolve`: TBD
+    - `smooth_est`: TBD
+    - `extrapolant`: TBD
+    - `controller`: TBD
```

Comment:
@ChrisRackauckas which of these need to go private for now?

Reply:
These should all be standard and documented? IIRC They are all already mentioned in the DifferentialEquations.jl docs.